src/network_engine.cpp

```diff
@@ -290,9 +290,9 @@ NetworkEngine::requestStep(Sp<Request> sreq)
         node.setExpired();
         if (not node.id)
             requests.erase(req.tid);
+        if (req.on_expired)
+            req.on_expired(req, false);
```

Comment:
At this point on_expired was already called (with true) through node.setExpired() -> req->setExpired().

Reply:
Thank you👍